geomesa-kafka/geomesa-kafka-datastore/src/main/scala/org/locationtech/geomesa/kafka/LiveKafkaConsumerFeatureSource.scala

```diff
@@ -114,14 +122,18 @@ class LiveKafkaConsumerFeatureSource(entry: ContentEntry,
     }, 0, 1, TimeUnit.SECONDS)
   }
 
-  override def run(): Unit = while (running.get) {
-    queue.take() match {
-      case update: CreateOrUpdate => featureCache.createOrUpdateFeature(update)
-      case del: Delete            => featureCache.removeFeature(del)
-      case clr: Clear             => featureCache.clear()
-      case m                      => throw new IllegalArgumentException(s"Unknown message: $m")
+  override def run(): Unit =
+    while (running.get) {
+      queue.take() match {
+        case update: CreateOrUpdate =>
+          featureCache.createOrUpdateFeature(update)
+          contentState.fireFeatureEvent(new KafkaFeatureEvent(this, Type.CHANGED, null, update.feature))
```

Comment:
Also, does ```fireFeatureEvent``` block?  Either we fire events asynchronously on a thread pool or we document it in such a way so that listeners ensure async.